alpha/engagements/2023/rust/update-2023-11.md

```diff
@@ -0,0 +1,57 @@
+
+### crates.io Request Failure Incident
+
+crates.io [experienced](https://status.crates.io/incidents/fsrmr4bk8trw) some failing requests and timeouts on October 18 due to an unexpected increase in traffic. While this looks to have been legitimate downloads requests spiking, Tobias is leading the investigation to ensure there is no major security issue.
+
+### Replace Ring with p256
+
+Tobias has [moved](https://github.com/rust-lang/crates.io/pull/7252) the crates.io dependency on `ring` over to `p256`. The `ring` project is releasing new versions without tags on the corresponding repository or even changelog entries, which is leaving us in the dark of what has changed or what is new. `p256` is a pure-Rust alternative for the only ring algorithm that we're currently using, and it also includes functions to parse the PEM format public keys.
+
+### Reading Dependencies from Cargo.toml
+
+To avoid the [hole](https://blog.vlt.sh/blog/the-massive-hole-in-the-npm-ecosystem) found in the `npm` ecosystem, Tobias is [proactively ensuring](https://github.com/rust-lang/crates.io/pull/7238) that all dependencies are read from an embedded cargo.toml file as opposed to a JSON metadata blog.
+
+
+### Typo-squatting Checks when new Packages are Published
+
+Building the on the release of [TypoMania](https://github.com/rustfoundation/typomania), Adam has now moved on to the next step by drafting a [PR](https://github.com/rust-lang/crates.io/pull/7206) to wire it into the crates.io publishing pipeline to notify Security Initiative team members when a potential typo-squatting incident occurs in real time when new crate packages are published.
+
+### Painter
+
+A [Painter](https://github.com/rustfoundation/painter) technical blog post is being drafted for publication in November. 
+
+Walter has added crate and function level data annotations, including for unsafe code. Also, Walter has implemented the parsing of 3rd party C-library dependencies in order to get visibility into potential vulnerabilities in libraries such as, for example, `libwebp`. If Walter is able to determine the version of the library, he can add the call graph, and see if any crates that import that C-library dependency are potentially vulnerable.
+
+## Community
+
+Walter and Joel attended an open source security round-table with members of the open source, language and package ecosystem community. Walter presented on [Painter](https://github.com/rustfoundation/painter) and we had folks wanting continuous access to its data - which we are working on setting up the infrastructure for that. Most of the presentations were on various mechanisms to secure package ecosystems, and there are ideas and implementations(e.g., machine-based identity, TUF, trusted publishers, etc.) that may make sense for the Rust Project.
+
+Adam and Tobias attended [PackagingCon 2023](https://packaging-con.org/) to network with other package manager ecosystems and share ideas on securing them.
```

Comment:
Don't know if you want to mention that I spoke. Don't know if _I_ want to mention that I spoke. :laughing:

Reply:
Yep. I will add a link to it. Thanks for the reminder. Hope the talk went well.